### ISSUE TITLE
Java14 Full Records Support Check Validation: ClassMemberImpliedModifierCheck

### ISSUE THREAD

**Issue by nmancus1** [checkstyle/checkstyle#8577]
Child of #8452 
Check documentation: https://checkstyle.sourceforge.io/config_modifier.html#ClassMemberImpliedModifier

Sorry, i've missed this issue.

This check should be updated. The `records` are always `static` as `enums` and `interfaces`:
```java
public class TestClass {
    int field;
    public record x() {
        public x {
            System.out.println(field); // Non-static variable field cannot be referenced from a static context
        }
    }
}
```
The test case should be
```java
public class TestClass {
    public static interface GoodInterface {} // OK
    public interface BadInterface {} // Implied modifier 'static' should be explicit. [ClassMemberImpliedModifier]

    public static enum GoodEnum {} // OK
    public enum BadEnum {} // Implied modifier 'static' should be explicit. [ClassMemberImpliedModifier]

    public static record GoodRecord() {} // OK
    public record BadRecord() {} // expected same violation
}
```
The check should be updated to support `TokenTypes.RECPRD_DEF`.
Also, a new property `violateImpliedStaticOnNestedRecord` (`true` by default) should be added.

_Originally posted by @pbludov in https://github.com/checkstyle/checkstyle/issues/8508#issuecomment-665197697_

CLI ouput:

```
➜  text-blocks-grammar /usr/lib/jvm/java-14-openjdk/bin/javac --enable-preview --source 14 TestClass.java
Note: TestClass.java uses preview language features.
Note: Recompile with -Xlint:preview for details.
➜  text-blocks-grammar cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="ClassMemberImpliedModifier"/>
    </module>
</module>
➜  text-blocks-grammar cat TestClass.java 
public class TestClass {
    public static interface GoodInterface {
    } // OK

    public interface BadInterface {
    } // Implied modifier 'static' should be explicit. [ClassMemberImpliedModifier]

    public static enum GoodEnum {} // OK

    public enum BadEnum {} // Implied modifier 'static' should be explicit. [ClassMemberImpliedModifier]

    public static record GoodRecord() {
    } // OK

    public record BadRecord() {
    } // expected same violation
}

➜  text-blocks-grammar java -jar /home/nick/IdeaProjects/checkstyle/target/checkstyle-8.36-SNAPSHOT-all.jar -c config.xml TestClass.java
Starting audit...
[ERROR] /home/nick/Desktop/text-blocks-grammar/TestClass.java:5:5: Implied modifier 'static' should be explicit. [ClassMemberImpliedModifier]
[ERROR] /home/nick/Desktop/text-blocks-grammar/TestClass.java:10:5: Implied modifier 'static' should be explicit. [ClassMemberImpliedModifier]
Audit done.
Checkstyle ends with 2 errors.

```
I would expect this check to report the missing `static` modifier for `BadRecord(){}`, but it does not.

**Comment by nmancus1** [comment]
Sorry, i've missed this issue.

This check should be updated. The `records` are always `static` as `enums` and `interfaces`:
```java
public class TestClass {
    int field;
    public record x() {
        public x {
            System.out.println(field); // Non-static variable field cannot be referenced from a static context
        }
    }
}
```
The test case should be
```java
public class TestClass {
    public static interface GoodInterface {} // OK
    public interface BadInterface {} // Implied modifier 'static' should be explicit. [ClassMemberImpliedModifier]

    public static enum GoodEnum {} // OK
    public enum BadEnum {} // Implied modifier 'static' should be explicit. [ClassMemberImpliedModifier]

    public static record GoodRecord() {} // OK
    public record BadRecord() {} // expected same violation
}
```
The check should be updated to support `TokenTypes.RECPRD_DEF`.
Also, a new property `violateImpliedStaticOnNestedRecord` (`true` by default) should be added.

_Originally posted by @pbludov in https://github.com/checkstyle/checkstyle/issues/8508#issuecomment-665197697_

**Comment by romani** [comment]
fix is merged.
